### PR TITLE
BlockPlacerCallback

### PR DESCRIPTION
**Pull request by nmacl**
Created a callback method to allow plugins to intercept the place method on Feature & Structure

https://github.com/SpongePowered/SpongeAPI/issues/2520

### CODE REVIEW

**Comment by Yeregorix** [comment]
I'm skeptical about using a callback instead of the blockchange event.
Also, the code is duplicated and doesn't even build.

**Comment by avaruus1** [comment]
> I'm skeptical about using a callback instead of the blockchange event. Also, the code is duplicated and doesn't even build.

Seems that there's also no PR for the implementation.

File: src/main/java/org/spongepowered/api/world/generation/feature/Feature.java [hunk 1]
```diff
@@ -55,22 +55,35 @@ public interface Feature extends DefaultedRegistryValue {
...
 
     /**
-     * Places the feature at given position and world
-     *
-     * @param world The world
-     * @param pos The position
-     *
-     * @return true when the feature was successfully placed
-     */
-    boolean place(ServerWorld world, Vector3i pos);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/world/generation/feature/Feature.java:65]: I don't see a reason why this couldn't remain, if a callback can be optionally an overloaded method.

**Comment by Yeregorix** [comment]
As I stated on the issue, the API already has everything needed, the real work is on the implementation.

**Pull request closed without merging**